core/libs/shared/src/lazy.rs

```diff
@@ -224,30 +256,36 @@ impl<T: Stagable> LazyTree<T> {
     }
 
     pub fn validate(&mut self) -> SharedResult<()> {
-        self.assert_no_orphans()?;
+        self.assert_all_files_decryptable()?;
         self.assert_only_folders_have_children()?;
         self.assert_no_cycles()?;
         self.assert_no_path_conflicts()?;
         // todo
         // self.assert_names_decryptable(account)?;
-        // share validations
+        self.assert_no_shared_links()?;
+        self.assert_no_duplicate_links()?;
+        self.assert_no_broken_links()?;
+        self.assert_no_owned_links()?;
```

Comment:
new tree validations for sharing (todo: corresponding merge conflict resolutions)